api/base/middleware.py

```diff
@@ -180,233 +139,3 @@ def process_response(self, request, response):
-        """
-        http://localhost:8000/preprints -> localhost
-        http://osf.io/preprints/... -> .osf.io
-        http://im-a-custom-domain.io/... -> .osf.io
-        http://staging-im-a-custom-domain.io/... -> .staging.osf.io
-
-
-        :param url:
-        :return:
-        """
-        if url.startswith('http://localhost:'):
-            return 'localhost'
-        else:
-            # for custom domains
-            if url.startswith('http://staging3') or url.startswith('https://staging3'):
-                return '.staging3.osf.io'
-            if url.startswith('http://staging2') or url.startswith('https://staging2'):
-                return '.staging2.osf.io'
-            if url.startswith('http://test') or url.startswith('https://test'):
-                return '.test.osf.io'
-            if url.startswith('http://staging') or url.startswith('https://staging'):
-                return '.staging.osf.io'
-            else:
-                return '.osf.io'
-
-    @staticmethod
-    def get_provider_from_url(referer_url: str) -> Optional[PreprintProvider]:
-        """
-        Takes the many preprint refer urls and try to figure out the provider based on that.
-        This will be eliminated post-sloan.
-        :param referer_url:
-        :return: PreprintProvider
-        """
-
-        # matches custom domains:
-        provider_domains = list(
-            PreprintProvider.objects.exclude(
-                domain='',
-            ).filter(
-                domain_redirect_enabled=True,  # must exclude our native domains like https://staging2.osf.io/
-            ).values_list(
-                'domain',
-                flat=True,
-            ),
-        )
-        provider_domains = [domains for domains in provider_domains if referer_url.startswith(domains)]
-
-        if provider_domains:
-            return PreprintProvider.objects.get(domain=provider_domains[0])
-
-        provider_ids_regex = '|'.join(
-            [re.escape(id) for id in PreprintProvider.objects.all().values_list('_id', flat=True)],
-        )
-        # matches:
-        # /ispp0  (preprint id)
-        path = urlparse(referer_url).path.replace('/', '')
-        preprint = Preprint.load(path)
-        if preprint:
-            return preprint.provider
-
-        # matches:
-        # /preprints
-        # /preprints/
-        # /preprints/notfound
-        # /preprints/foorxiv
-        # /preprints/foorxiv/
-        # /preprints/foorxiv/guid0
-        provider_regex = r'preprints($|\/$|\/(?P<provider_id>{})|)'.format(provider_ids_regex)
-        match = re.match(re.escape(DOMAIN) + provider_regex, referer_url)
-        if match:
-            provider_id = match.groupdict().get('provider_id')
-            if provider_id:
-                return PreprintProvider.objects.get(_id=provider_id)
-            return PreprintProvider.objects.get(_id='osf')
-
-    @staticmethod
-    def override_flag_activity(sloan_flag_name, waffles_data, user):
```

Comment:
With is for the case where an "anonymous" user logs in changing their A/B status.